Cargo.toml

```diff
@@ -42,7 +42,7 @@ serde_json = { version = "1.0.79", default-features = false }
 url = { version = "2.0.0", optional = true }
 
 # websocket dependent features
-jsonrpsee = { version = "0.16", optional = true, features = ["async-client", "client-ws-transport", "jsonrpsee-types"] }
+jsonrpsee = { version = "0.21", optional = true, features = ["async-client", "client-ws-transport-native-tls", "jsonrpsee-types"] }
```

Comment:
This is less on this PR but a general question: It seems wrong to me, that we explicitly reference the different websocket libraries here. This does mean, that I have to include tungstenite, even when I'm using the api client with jsonrpsee. Is this correct? 

On the specific code here: What if I want to use the api client with jsonrpsee, but without the native-tls feature? As far as I understand, there's no reason to not allow this. But it seems to be prevented by this.

Reply:
>This does mean, that I have to include tungstenite, even when I'm using the api client with jsonrpsee. Is this correct?

No, that is not correct. The dependency imports of the rpc-clients is optional and only included if the correspondent feature is chosen.

> What if I want to use the api client with jsonrpsee, but without the native-tls feature? As far as I understand, there's no reason to not allow this. But it seems to be prevented by this.

Do you have something in mind? We could forward the features of the jsonrpsee client, but that would mean more features on our side, I believe… or is there a way around it?
Or we could allow that the user constructs his/her own jsonrpsee client and just wraps our wrapper struct around it: https://github.com/scs/substrate-api-client/blob/master/src/rpc/jsonrpsee_client/mod.rs#L30-L33:
```rust
// Something like this:
let jsonrpsee_client = ClientBuilder::default()
		.max_buffer_capacity_per_subscription(4096)
		.build_with_tokio(tx, rx);
let api_client = JsonrpseeClient { inner: Arc::new(client) };
```

Reply:
Actually, I'm not sure if my first approach would work as long as we are constructing our own jsonrpsee-client.